Installation issue: proj build_system=autotools

### Steps to reproduce the issue

```console
$ spack install proj build_system=autotools
```


### Error message

<details><summary>Error message</summary><pre>
1 error found in build log:
     150    checking for SQLITE3... yes
     151    checking for sqlite3... yes
     152    checking for TIFF... yes
     153            found libcurl version 8.0.1
     154    checking for curl_global_init in -lcurl... yes
     155    using external GTest.
  >> 156    checking for GTEST... configure: error: Package requirements (gtest >= 1.8.0) were not met:
     157    
     158    No package 'gtest' found
     159    
     160    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     161    installed software in a non-standard prefix.
</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (74ad92e16b8a5e56152e75d264b99f006763586b)
* **Python:** 3.9.2
* **Platform:** linux-debian11-skylake
* **Concretizer:** clingo

### Additional information

@adamjstewart I don't think that uploading any files is required here (sorry, I have to lie below by setting the corresponding checkbox). It just looks like the configure script fails if GTest is not available, which is the case because `googletest` is a `test` dependency.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/proj/package.py
+++ b/var/spack/repos/builtin/packages/proj/package.py
@@ -87,7 +87,7 @@ class Proj(CMakePackage, AutotoolsPackage):
     depends_on("sqlite@3.11:", when="@6:")
     depends_on("libtiff@4:", when="@7:+tiff")
     depends_on("curl@7.29:", when="@7:+curl")
-    depends_on("googletest", when="@6:", type="test")
+    depends_on("googletest@1.8:", when="@6:", type="test")
 
     build_system(
         conditional("autotools", when="@:8"), conditional("cmake", when="@5:"), default="cmake"
@@ -132,7 +132,7 @@ class AutotoolsBuilder(autotools.AutotoolsBuilder, BaseBuilder):
     def configure_args(self):
         args = []
 
-        if self.spec.satisfies("@6:"):
+        if self.spec.satisfies("@6:") and self.pkg.run_tests:
             args.append("--with-external-gtest")
 
         if self.spec.satisfies("@7:"):

